src/main/java/seedu/address/logic/commands/ingredientcommands/IngredientResetAllCommand.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands.ingredientcommands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ingredient.Amount;
+import seedu.address.model.ingredient.Ingredient;
+
+/**
+ * Set the level of one specific ingredient to a specific level.
+ */
+public class IngredientResetAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "i-reset-all";
+    public static final String RESET_AMOUNT = "0";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :reset all the ingredient levels in tCheck."
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "All ingredient levels are reset to 0.";
+    public static final String MESSAGE_NO_CHANGE = "All Ingredient levels are already at 0.";
+
+    /**
+     * Constructs an IngredientResetAll command.
+     */
+    public IngredientResetAllCommand() {
+        super();
+    }
+
+    /**
+     * Executes the command and returns the result message.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
+     * @throws CommandException If an error occurs during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Ingredient> lastShownList = model.getFilteredIngredientList();
+
+        boolean isNotAlreadyReset = false;
```

Comment:
I think that perhaps naming the boolean `isReset` is better, so that there isn't a double negative. This is just a grammar comment.

Reply:
I would like to highlight that it is not "reset" before this execution, so I thought it is clearer than isReset..?

Reply:
I see,, then it makes sense to use `isNotAlreadyReset`.